cephfs/admin/fs_quiesce.go

```diff
@@ -0,0 +1,131 @@
+//go:build ceph_preview
+
+package admin
+
+import "fmt"
+
+// FixedPointFloat is a custom type that implements the MarshalJSON interface.
+// This is used to format float64 values to two decimal places.
+// By default these get converted to integers in the JSON output and
+// fail the command.
+type FixedPointFloat float64
+
+// MarshalJSON provides a custom implementation for the JSON marshalling
+// of FixedPointFloat. It formats the float to two decimal places.
+func (fpf FixedPointFloat) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("%.2f", float64(fpf))), nil
+}
+
+// fsQuiesceFields is the internal type used to create JSON for ceph.
+// See FsQuiesceOptions for the type that users of the library
+// interact with.
+type fsQuiesceFields struct {
+	Prefix     string          `json:"prefix"`
+	VolName    string          `json:"vol_name"`
+	GroupName  string          `json:"group_name,omitempty"`
+	Members    []string        `json:"members,omitempty"`
+	SetId      string          `json:"set_id,omitempty"`
+	Timeout    FixedPointFloat `json:"timeout,omitempty"`
+	Expiration FixedPointFloat `json:"expiration,omitempty"`
+	AwaitFor   FixedPointFloat `json:"await_for,omitempty"`
+	Await      bool            `json:"await,omitempty"`
+	IfVersion  int             `json:"if_version,omitempty"`
+	Include    bool            `json:"include,omitempty"`
+	Exclude    bool            `json:"exclude,omitempty"`
+	Reset      bool            `json:"reset,omitempty"`
+	Release    bool            `json:"release,omitempty"`
+	Query      bool            `json:"query,omitempty"`
+	All        bool            `json:"all,omitempty"`
+	Cancel     bool            `json:"cancel,omitempty"`
+}
+
+// SubVolumeQuiesceOptions are used to specify optional, non-identifying, values
+// to be used when quiescing a subvolume.
+type FsQuiesceOptions struct {
+	Timeout    FixedPointFloat
+	Expiration FixedPointFloat
+	AwaitFor   FixedPointFloat
+	Await      bool
+	IfVersion  int
+	Include    bool
+	Exclude    bool
+	Reset      bool
+	Release    bool
+	Query      bool
+	All        bool
+	Cancel     bool
+}
+
+// toFields is used to convert the FsQuiesceOptions to the internal
+// fsQuiesceFields type.
+func (o *FsQuiesceOptions) toFields(volume, group string, subvolumes []string, SetId string) *fsQuiesceFields {
+	return &fsQuiesceFields{
+		Prefix:     "fs quiesce",
+		VolName:    volume,
+		GroupName:  group,
+		Members:    subvolumes,
+		SetId:      SetId,
+		Timeout:    o.Timeout,
```

Comment:
here you may need to convert `o.Timeout` to a `float64` then, with `float64(o.Timeout)`

Reply:
Why?  FsQuiesceOptions.Timeout is fixedPointFloat and so is fsQuiesceFields.Timeout.